lib/core/network/protocols/http+websocket.js

```diff
@@ -0,0 +1,910 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2020 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const querystring = require('querystring');
+const url = require('url');
+const zlib = require('zlib');
+
+const uWS = require('uWebSockets.js');
+
+const { Request } = require('../../../api/request');
+const { KuzzleError } = require('../../../kerror/errors');
+const Protocol = require('./protocol');
+const ClientConnection = require('../clientConnection');
+const removeErrorStack = require('../removeErrorStack');
+const debug = require('../../../util/debug');
+const kerror = require('../../../kerror');
+const HttpMessage = require('./httpMessage');
+
+const kerrorWS = kerror.wrap('network', 'websocket');
+const kerrorHTTP = kerror.wrap('network', 'http');
+const debugWS = debug('kuzzle:network:protocols:websocket');
+const debugHTTP = debug('kuzzle:network:protocols:http');
+
+// The idleTimeout option should never be deactivated, so instead we use
+// a default value for backward-compatibility
+const DEFAULT_IDLE_TIMEOUT = 60000;
+
+// Size of backpressure an individual socket can handle before needing to drain
+const WS_MAX_BACKPRESSURE = 4096;
```

Comment:
What does this 4096 mean?
`Back-pressure` as = `size`?

Reply:
It's the size in bytes, I'll amend the comment.

Backpressure is the right term for it though, but it should be the backpressure buffer size: since websocket is on top of TCP, the server expects an ACK packet each time it sends something to a client. If the client fails to send an acknowledgement, or if it is too slow, or if there are network problems, then the server HAS to keep the payload it needs to send in a buffer. This is backpressure: a load imposed to something, because of something else (here, the server has to keep things in memory because of client sockets).

If we want to be exact about the name of the constant, then it should be `WS_MAX_BACKPRESSURE_BUFFER_SIZE`, but it's a bit long to my taste.